### PR TITLE
TAD Export 2 ApplicationsByDemographicDomicileAndDegreeClassExport

### PR DESCRIPTION
**Pull request by george-bit**
## Context
This one of the data exports requested by TAD, should contain break down of application choices by age group, sex, ethnicity, disability, domicile, degree class.

## Changes proposed in this pull request

This PR adds the export to the Support interface and exposes it on the data API

## Guidance to review

Let me know if this looks okay

## Link to Trello card

https://trello.com/c/BbRYwaRr/4085-tad-export-2-age-ed-domicile-degree-class-and-hesa-data-by-outcome

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by stevehook — COMMENTED** [review]
Tested locally and all looks good. The code looks really tidy and easy to follow (despite being a complicated feature). I've left a few questions inline about things I'm not sure about.

File: app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    end
+
+    def domicile_sql
+      "WHEN f.country = 'GB' THEN 'UK'
+      WHEN f.country IN (#{EU_COUNTRY_CODES.map { |country| "'#{country}'" }.join(',')}) THEN 'EU'
+      ELSE
+     'Non EU'"
+    end
+
+    def degree_class_sql
```
Comment (davidgisbey) [inline app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb:188]: I think we can probably use the aggr syntax here to deal with retuning the highest grade here 

 `"WHEN 'First class honours'  = ANY(ARRAY_AGG(q.grade)) THEN ARRAY['0', 'First class honours']`

File: app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                  #{domicile_sql}
+                END domicile
+            FROM
+                application_forms f
+            LEFT JOIN
+                candidates c ON f.candidate_id = c.id
+            LEFT JOIN
+                application_choices ch ON ch.application_form_id = f.id
+            LEFT JOIN
+              application_qualifications q ON q.application_form_id = f.id
```
[the remark this answers is not in the document]
Comment (davidgisbey) [reply app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb:119]: left a comment on your first comment below

File: app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      DataExporter.perform_async(SupportInterface::ApplicationsByDemographicDomicileAndDegreeClassExport, data_export.id)
+    end
+
+    def call(*)
+      results = ActiveRecord::Base
+        .connection
+        .execute(query)
+        .to_a
+        .reject { |row| row.values.include?(nil) }
+        .reject { |row| row.values.include?('[]') }
```
[the remark this answers is not in the document]
Comment (george-bit) [reply app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb:17]: I think this is to remove any blank degree grades for the first row and any empty arrays for the diability column. Will have a look at refactoring into the query as suggested 👍

File: app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (george-bit) [reply app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb:119]: 👍 to all the above ☝️

File: app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb (hunk 1 above)
Comment (george-bit) [reply app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb:191]: Managed to resolve the `MIN(raw_data.degree_class[1]) AS degree_class` on line 143. couldn't figure out how to get it to work without assigning the values to an array with multiple elements. 🤔

**Review by stevehook — APPROVED** [review]
Tested locally again and generates the data I'd expect. Great stuff!

**Pull request merged**